spec/inputs/syslog_spec.rb

```diff
@@ -68,11 +68,56 @@
 
       insist { events.length } == event_count
       event_count.times do |i|
-        insist { events[i]["tags"] } == ["_grokparsefailure_sysloginputplugin"]
+        insist { events[i]["tags"] } == ["_grokparsefailure_sysloginput"]
+      end
+    end
+  end
+
+  describe "should properly handle locale and timezone" do
+    port = 5511
+    event_count = 10
+
+    config <<-CONFIG
+      input {
+        syslog {
+          type => "blah"
+          port => #{port}
+          locale => "en"
+          timezone => "UTC"
+        }
+      }
+    CONFIG
+
+    input do |pipeline, queue|
+      Thread.new { pipeline.run }
+      sleep 0.1 while !pipeline.ready?
+
+      socket = Stud.try(5.times) { TCPSocket.new("127.0.0.1", port) }
+      event_count.times do |i|
+        socket.puts("<164>Oct 26 15:19:25 1.2.3.4 %ASA-4-106023: Deny udp src DRAC:10.1.2.3/43434 dst outside:192.168.0.1/53 by access-group \"acl_drac\" [0x0, 0x0]")
+      end
+      socket.close
+
+      events = event_count.times.collect { queue.pop }
+
+      insist { events.length } == event_count
+      events.each do |event|
+        insist { event["@timestamp"].to_iso8601 } == "#{Time.now.year}-10-26T15:19:25.000Z"
       end
     end
   end
```

Comment:
Missing a test if we don't define the `locale` option, not sure how the date parser will work if we send `nil`